### PR TITLE
Add missing `f` for fstring and update imports

### PR DESCRIPTION
**Pull request by beeankha**
Resolves [this comment](https://github.com/conda/conda/pull/12475#issuecomment-1552767334).

### CODE REVIEW

**Review by sven6002 — DISMISSED** [review]
Looks good to me..

File: conda/gateways/repodata/__init__.py [hunk 1]
```diff
@@ -513,7 +512,7 @@ def __setitem__(self, key: str, item: Any) -> None:
         if key in self._aliased:
             key = key[1:]  # strip underscore
         if key in self._strings and not isinstance(item, str):
-            warnings.warn('Replaced non-str RepodataState[{key}] with ""')
+            warnings.warn(f'Replaced non-str RepodataState[{key}] with ""')
```
Comment (jaimergp) [inline conda/gateways/repodata/__init__.py:515]: Two things:

- Should we also log the `item` that got replaced?
- Should this be `logging.warning` instead?

Comment (beeankha) [reply conda/gateways/repodata/__init__.py]: I consulted with @sven6002 and @kenodegard and we think it makes more sense for now to only use the `warnings` module exclusively for deprecations and `logging` everywhere else. The code has been updated accordingly!

**Comment by jaimergp** [comment]
> Change the way we warn

Sounds like the catchy phrase at the end of a song 😂

**Review by jaimergp — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**